ddtrace/debugging/_capture/utils.py

```diff
@@ -140,14 +141,20 @@ def capture_exc_info(exc_info):
     }
 
 
-def capture_value(value, level=MAXLEVEL, maxlen=MAXLEN, maxsize=MAXSIZE, maxfields=MAXFIELDS):
-    # type: (Any, int, int, int, int) -> Dict[str, Any]
+def capture_value(value, level=MAXLEVEL, maxlen=MAXLEN, maxsize=MAXSIZE, maxfields=MAXFIELDS, hg=None):
+    # type: (Any, int, int, int, int, Optional[HourGlass]) -> Dict[str, Any]
     _type = type(value)
 
     if _type in BUILTIN_SIMPLE_TYPES:
         if _type is NoneType:
             return {"type": "NoneType", "isNull": True}
 
+        if hg is not None and not hg.trickling():
+            return {
```

Comment:
Maybe there is some room to reduce code duplication.

1. have utility function create_capture_value* that generate that json like create_capture_value_not_captured(_type, "timeout", [size]).
2. make those string as constant NOT_CAPTURED_REASON = "notCapturedReason"

Reply:
I think these structures are simple enough that we can avoid the cost of an extra function call and a global variable lookup. WDYT?

Reply:
I don't like magic strings duplicated in multiple places in the code. 
However, this is only file we use those and I don't think this is going to chance any time soon. 
So, I will trust your judgement on what is more maintainable.